rise/static/main.js

```diff
@@ -583,6 +579,66 @@ function reveal_cell_index(notebook) {
     return result;
 }
 
+function registerHelperActions() {
+
+    function init_metadata_slideshow(optional_cell) {
+      // use selected cell if not specified
+      var cell = optional_cell || Jupyter.notebook.get_selected_cell();
+      let metadata = cell.metadata;
+      if (metadata.slideshow === undefined)
+        metadata.slideshow = {};
+      return metadata.slideshow;
+    }
+    
+    // accessing Jupyter.actions directly results in a warning message
+    // https://github.com/jupyter/notebook/issues/2401
+    let actions = Jupyter.notebook.keyboard_manager.actions;
+    actions.register(
+      {
+        help: '(un)set current cell as a Slide cell',
+        handler: function() {
+          let slideshow = init_metadata_slideshow();
+          slideshow.slide_type = (slideshow.slide_type == 'slide') ? '' : 'slide';
+	  Jupyter.CellToolbar.rebuild_all();
+        }
+      },
+      "toggle-slide", "RISE");
+
+    actions.register(
+      {
+        help: '(un)set current cell as a Fragment cell',
+        handler: function() {
+          let slideshow = init_metadata_slideshow();
+          slideshow.slide_type = (slideshow.slide_type == 'fragment') ? '' : 'fragment';
+	  Jupyter.CellToolbar.rebuild_all();
+        }
+      },
+      "toggle-fragment", "RISE");
+
+    actions.register(
+      { help: 'render all cells (all cells go to command mode)',
+        handler: function() {
+          Jupyter.notebook.get_cells().forEach(function(cell){
+	    cell.render();
+          })
+        }
+      },
+      "render-all-cells", "RISE");
+
+    actions.register(
+      {
+        help: 'edit all cells (all cells go to edit mode)',
+        handler: function() {
+          Jupyter.notebook.get_cells().forEach(function(cell){
+	    cell.unrender();
+          })
+        }
+      },
+      "edit-all-cells", "RISE");
```

Comment:
In which use cases render-all-cells and edit-all-cells are useful?

Reply:
well, in presentation mode, having a one-key shortcut to so render-all-cells is really convenient for all the cases where you just made a few changes in your cells and forgot to render them properly.

the other way around (edit-all-cells), it is less obvious; I use this much more occasionnally to be honest

Reply:
Yes, my feeling was the same, render-all-cells seems like something frequently used, but not for edit-all-cells... still, I like the symmetry having both actions, so I am OK to leave it as is.

Reply:
As I said before I will keep this two but I am not assigning them any default shortcut. It is up to the user to configure it if they are interested, at least for now.